components/expression/expression-panel.js

```diff
@@ -68,7 +71,54 @@ genotet.ExpressionPanel.prototype.initPanel = function() {
           type: bSwitch.type
         });
       }.bind(this));
-   }, this);
+  }, this);
+
+  // Add and remove gene profiles
+  this.selectProfiles_
+    .on('select2:select', function(event) {
+      var geneIndex = event.params.data.element.index;
+      this.signal('addGeneProfile', geneIndex);
+    }.bind(this))
+    .on('select2:unselect', function(event) {
+      var geneIndex = event.params.data.element.index;
+      this.signal('removeGeneProfile', geneIndex);
+    }.bind(this));
+
+  // Gene update
+  ['setGene', 'addGene', 'removeGene'].forEach(function(method) {
+    this.container_.find('#genes #' + method).click(function() {
+      var input = this.container_.find('#genes input');
+      var geneRegex = input.val();
+      if (geneRegex == '') {
+        genotet.warning('missing input gene selection')
+        return;
+      }
+      input.val('');
+      this.signal('update', {
+        type: 'gene',
+        regex: geneRegex,
+        method: method
+      });
+    }.bind(this));
+  }, this);
+
+  // Condition update
+  ['setCondition', 'addCondition', 'removeCondition'].forEach(function(method) {
```

Comment:
Actually I don't think merge them is a good way. Because they are much more different at selection names. Merge would cost more codes to check whether to revise gene or condition.